pkg/controller/render/render_controller.go

```diff
@@ -416,6 +460,76 @@ func (ctrl *Controller) syncMachineConfigPool(key string) error {
 		return nil
 	}
 
+	// If it's a pool named "layered", it's special, designate it automatically
+	// TODO(jkyros): take this out when we decide actual UX
+	if pool.Name == "layered" {
+		if pool.Labels == nil {
+			pool.Labels = map[string]string{}
+		}
+		pool.Labels[ctrlcommon.ExperimentalLayeringPoolLabel] = ""
+	}
+
+	// If this is is a layered pool check to see if we have an image stream
+	// If we do, snarf out the latest image and annotate the pool with it
+	if ctrlcommon.IsLayeredPool(pool) {
+
+		// TODO(jkyros): This will probably requeue the pool because config changed, we should probably do this elsewhere
+		ctrl.experimentalAddEntitlements(pool)
+
+		// We don't want to crash if these are empty
+		if pool.Annotations == nil {
+			pool.Annotations = map[string]string{}
+		}
+		// Do we have an image stream for this pool?
+		is, err := ctrl.imageclient.ImageV1().ImageStreams(ctrlcommon.MCONamespace).Get(context.TODO(), pool.Name+ctrlcommon.ImageStreamSuffixMcoContent, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			glog.Warningf("Image stream for %s does not exist (yet?): %s", pool.Name, err)
+		} else {
+
+			// If we do, grab the latest tag
+			for _, tag := range is.Status.Tags {
+				for _, image := range tag.Items {
+
+					// If this is different than our current tag, grab it and annotate the pool
+					if imageReference, ok := pool.Annotations[ctrlcommon.ExperimentalNewestLayeredImageAnnotationKey]; !ok || imageReference != image.DockerImageReference {
+						glog.Infof("imagestream %s newest is: %s", is.Name, image.DockerImageReference)
+						// Annotate the pool with its most recent image so node-controller can use it
+						pool.Annotations[ctrlcommon.ExperimentalNewestLayeredImageAnnotationKey] = image.DockerImageReference
+
+						// get the actual image so we can read its labels
+						fullImage, err := ctrl.imageclient.ImageV1().Images().Get(context.TODO(), image.Image, metav1.GetOptions{})
+						if err != nil {
+							return err
+						}
+
+						// We need the labels out of the docker image but it's a raw extension
+						dockerLabels := struct {
+							Config struct {
+								Labels map[string]string `json:"Labels"`
+							} `json:"Config"`
+						}{}
+
+						// Get the labels out and see what config this is
+						err = json.Unmarshal(fullImage.DockerImageMetadata.Raw, &dockerLabels)
+						if err != nil {
+							glog.Warningf("Could not get labels from docker image metadata: %s", err)
+						}
+
+						// Tag what config this came from so we know it's the right image
```

Comment:
This would go away if the pool's target *is* the image right?

Reply:
Correct, this is only for where we are right now where the image is the "vehicle that gets us to the proper rendered-config"